File: backend/app/schemas/audit.py
"""Audit log read schema."""
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class AuditLogRead(BaseModel):
    id: UUID
    action: str
    detail: Optional[str] = None
    user_id: Optional[UUID] = None
    created_on: datetime

    model_config = {"from_attributes": True}

File: frontend/src/lib/audit.ts
import { apiFetch } from "./api";
import type { Page } from "./pagination";

export interface AuditEntry {
  id: string;
  action: string;
  detail?: string | null;
  user_id?: string | null;
  created_on: string;
}

export const listAuditLogs = () => apiFetch<Page<AuditEntry>>("/audit-logs");
